src/client/utils/tracking/trackOfferEvent.ts

```diff
@@ -100,62 +105,51 @@ export const useTrackOfferEvent = () => {
 
   const [selectedInsuranceTypes] = useSelectedInsuranceTypes()
 
-  const { data: quoteCartQueryData } = useQuoteCartQuery({
-    variables: {
-      id: quoteCartId,
-      locale: isoLocale,
-    },
-  })
-
-  const isReferralCodeUsed =
-    getMonthlyCostDeductionIncentive(quoteCartQueryData) !== undefined
-
-  const selectedBundleVariant = getSelectedBundleVariant(
-    quoteCartQueryData,
-    selectedInsuranceTypes,
-  )
-
-  const [eventQueue, setEventQueue] = useState<EventParameters[]>([])
-
-  const trackOfferCallback = useCallback(
-    (
-      { eventName, options = {} }: EventParameters,
-      selectedBundleVariant: QuoteBundleVariant,
-    ) => {
-      trackOfferEvent(
-        eventName,
-        selectedBundleVariant.bundle,
-        isReferralCodeUsed,
-        {
-          quoteCartId,
-          ...options,
-        },
-      )
-    },
-    [isReferralCodeUsed, quoteCartId],
-  )
-
   const trackOfferHandler = useCallback(
-    (eventParams: EventParameters) => {
-      if (selectedBundleVariant) {
-        trackOfferCallback(eventParams, selectedBundleVariant)
-      } else {
-        setEventQueue((prevQueue) => [...prevQueue, eventParams])
+    ({ eventName, options = {} }: EventParameters) => {
+      const runQuery = async () => {
+        if (apolloClient && quoteCartId) {
+          try {
+            const quoteCartQuery = await apolloClient!.client.query<
```

Comment:
Why do we need the hashbang here if we do the if statement above?

Reply:
@marian-buc bump ;)

Reply:
https://github.com/HedvigInsurance/web-onboarding/blob/0c888ab09114125b3ad47a5b7c76724a5e11d461/src/client/apolloClient.ts#L89

Query still might fail event if quoteCartId is supplied

Reply:
hmmm not sure what you mean there.. I was referring to the non-null assertion operator `!` here:

```
apolloClient!.client.query
```

Why is it still necessary? :)

Reply:
haha, yup, removed it in the new PR as well
[by passing the client as a param](https://github.com/HedvigInsurance/web-onboarding/pull/972/commits/9e89fe0c8438dd94f416448a38a84a8927907c26#diff-baa1c36a82a2c7876dc53d4df01709b66f9257464f8359b7f700d8e628abe6a5R32)